test/functional/rpc_packages.py

```diff
@@ -49,94 +39,94 @@ def assert_testres_equal(self, package_hex, testres_expected):
         assert_equal(shuffled_testres, self.nodes[0].testmempoolaccept(shuffled_package))
 
     def run_test(self):
-        self.log.info("Generate blocks to create UTXOs")
         node = self.nodes[0]
-        self.privkeys = [node.get_deterministic_priv_key().key]
-        self.address = node.get_deterministic_priv_key().address
-        self.coins = []
-        # The last 100 coinbase transactions are premature
-        for b in self.generatetoaddress(node, 220, self.address)[:-100]:
-            coinbase = node.getblock(blockhash=b, verbosity=2)["tx"][0]
-            self.coins.append({
+
+        # get an UTXO that requires signature to be spent
+        deterministic_address = node.get_deterministic_priv_key().address
+        blockhash = self.generatetoaddress(node, 1, deterministic_address)[0]
+        coinbase = node.getblock(blockhash=blockhash, verbosity=2)["tx"][0]
+        coin = {
                 "txid": coinbase["txid"],
-                "amount": coinbase["vout"][0]["value"],
+                "value": coinbase["vout"][0]["value"],
                 "scriptPubKey": coinbase["vout"][0]["scriptPubKey"],
-            })
+                "vout": 0,
+                "height": 0
+            }
```

Comment:
There's no reason not to construct this coin inside `test_independent`. You can use `node.getblock(blockhash=node.getblockhash(height=1), verbosity=2)["tx"][0]` to get the coinbase from that specific block, but also why not just use create self transfer?

Reply:
`node.getblock(blockhash=node.getblockhash(height=1), verbosity=2)["tx"][0]` returns a UTXO that does not need a signature, as the UTXO are being generated for a `ADDRESS_OP_TRUE` miniwallet.

The same applies to `create_self_transfer`.

The reason why this code is inside `test_independent` is https://github.com/bitcoin/bitcoin/pull/25986#discussion_r963868741.